source/tearable-atomics.bs

```diff
@@ -96,13 +97,67 @@ A more in-depth
 [discussion of seqlock](http://safari.ece.cmu.edu/MSPC2012/slides_posters/boehm-slides.pdf)
 is available.
 
-For the purpose of our discussion, it is especially interesting to consides
+For the purpose of our discussion, it is especially interesting to considers
 value types `T` which are never lock-free.
 
 Work-Stealing Deque {#wsdeque}
 -------------------
 
-Issue: FIXME Billy! Include some code, and explain speculation.
+It appears intended that implementations of the parallelism TS back scheduling
+of partitioned work using an ABP work-stealing scheduler, originally described
+in ["Thread Scheduling for Multiprogrammed Microprocessors" by Nimar S. Arora,
+Robert D. Blumofe, and C. Greg Plaxton](https://www.eecis.udel.edu/~cavazos/cisc879-spring2008/papers/arora98thread.pdf).
+Under this model, each thread has a local deque of work where it can access the
+"bottom" of the deque without any synchronization overhead, but other threads
+can concurrently remove work from the "top". A similar data structure was
+presented implementable on hardware without double-wide CAS instructions in
+["Dynamic Circular Work-Stealing Deque" by David Chase and Yossi Lev](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.170.1097&rep=rep1&type=pdf).
+
+In the Chase-Lev deque, the "top" counter is used to track concurrent access to
+the top of the deque; and access to the actual elements in the top of the deque
+is unsynchronized. From the original paper:
+
+    public Object steal() {
```

Comment:
You'll want to wrap all code in `<xmp>` ... `</xmp>` as above. It'll format and highlight better than regular markdown, and then you don't need to HTML escape `<`, `>`, and `&`.